rust/tw_bitcoin/src/ordinals.rs

```
@@ -0,0 +1,162 @@
+    use bitcoin::opcodes::all::*;
+    use bitcoin::opcodes::*;
+
+    // TODO: Check overflow
+    let (op_push, size_buf) = get_op_push(mime.len() as u32)?;
+
+    // Prepare content-type buffer.
+    let mut mime_buf = PushBytesBuf::new();
+
+    // For any sizes below 75, we use encode as `OP_PUSHBYTES[0|1|2|...|75]`.
+    // Fany any sized above 75, we use encode as `OP_PUSHDATA[1|2|3|4] <SIZE_BUF>`.
+    mime_buf.push(op_push.to_u8()).unwrap();
+    if let Some(size_buf) = size_buf {
+        mime_buf.extend_from_slice(&size_buf).unwrap();
+    }
+    mime_buf.extend_from_slice(mime).unwrap();
+
+    // Prepare data buffer.
+    let mut data_buf = PushBytesBuf::new();
+    let (op_push, size_buf) = get_op_push(data.len() as u32)?;
+    data_buf.push(op_push.to_u8()).unwrap();
+    if let Some(size_buf) = size_buf {
+        data_buf.extend_from_slice(&size_buf).unwrap();
+    }
+    data_buf.extend_from_slice(data).unwrap();
+
+    // Create an Ordinals Inscription.
+    let script = ScriptBuf::builder()
+        .push_opcode(OP_FALSE)
+        .push_opcode(OP_IF)
+        .push_opcode(OP_PUSHBYTES_3)
+        .push_slice(b"ord")
+        .push_opcode(OP_PUSHNUM_1)
+        .push_slice(mime_buf)
+        .push_opcode(OP_PUSHBYTES_0)
+        .push_slice(data_buf)
+        .push_opcode(OP_ENDIF)
+        .into_script();
+
+    dbg!(&script);
+
+    // Generate the necessary spending information. As mentioned in the
+    // documentation of this function at the top, this serves two purposes;
+    // setting the spending condition and actually claiming the spending
+    // condition.
+    let spend_info = TaprootBuilder::new()
+        .add_leaf(0, script.clone())
+        .map_err(|_| Error::Todo)?
+        .finalize(
+            &secp256k1::Secp256k1::new(),
+            XOnlyPublicKey::from(internal_key.inner),
+        )
+        //.map_err(|_| Error::Todo)?;
+        .unwrap();
```

Maybe return an error?